scalability_and_performance/cnv-performance-scale-resilience-practices/recommended-node-practices.adoc

```diff
@@ -0,0 +1,86 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="recommended-node-practices"]
+= Recommended node practices
+include::_attributes/common-attributes.adoc[]
+:context: recommended-node-practices
+
+toc::[]
+
+This topic provides recommended node practices for a Container Native Virtualization environment to be performant, scalable and resilient in {product-title}.
+
+If a node fails and Node Health Checks operator is not deployed  and configured with an appropriate remediation operator - Self Node Remediation (SNR) and/or Fence Agents Remediation (FAR) on your cluster, Virtual Machines (VMs) will encounter downtime until the node recovers as they are not automatically recovered to other available nodes by default.
+
+Here are the configurations recommended to avoid virtual machines (VMs) downtime on a critical environment and for your workloads to be resilient under node failure conditions:
+
+-  Configure virtual machines (VMs) with `runStrategy: Always` to allow the recovery to other available nodes.
+
+include::modules/virt-configuring-runstrategy-vm.adoc[leveloffset=+2]
+
+
+- Enable Node Health Checks to monitor the node status change from Ready to Unknown or NotReady, when a given node has not been able to contact the control plane for a defined period of time. spec.unhealthyConditions[].duration can be tuned to reduce the recovery timing. For example:
+
+```
+apiVersion: remediation.medik8s.io/v1alpha1
+kind: NodeHealthCheck
+metadata:
+  name: nodehealthcheck-sample
+spec:
+  minHealthy: <set to 1 for critical environments> 
+  pauseRequests: 
+    - <pause-test-cluster>
+  remediationTemplate: 
+    apiVersion: self-node-remediation.medik8s.io/v1alpha1
+    name: self-node-remediation-automatic-strategy-template
+    namespace: openshift-workload-availability
+    kind: SelfNodeRemediationTemplate
+  escalatingRemediations: 
+    - remediationTemplate:
+        apiVersion: self-node-remediation.medik8s.io/v1alpha1
+        name: self-node-remediation-resource-deletion-template
+        namespace: openshift-workload-availability
+        kind: SelfNodeRemediationTemplate
+    order: 1
+    timeout: 300s
+  selector: 
+    matchExpressions:
+      - key: node-role.kubernetes.io/worker
+        operator: Exists
+  unhealthyConditions: 
+    - type: Ready
+      status: "False"
+      duration: <set to lower timings for faster recovery, for example 20s> 
+    - type: Ready
+      status: Unknown
+      duration: <set to lower timings for faster recovery, for example 20s>
+```
+
+Node Health Check operator supports Self Node Remediation (SNR) or Fence Agents Remediation (FAR) as the remediation operator to apply after detecting a node failure.
```

Comment:
NHC could more remediation operators than SNR or FAR, we just tried and tested those. I would clarify this.